Sources/MapboxCoreNavigation/NavigationProvider.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import MapboxDirections
+
+/**
+ Protocol which defines a type which can be used for fetching or refreshing routes.
+ 
+ SDK provides conformance to this protocol for `Directions` and `NavigationRouter`.
+ */
+public protocol NavigationProvider {
```

Comment:
The name of this protocol says that an implementation of this protocol “provides navigation”. That seems a bit too abstract. I’d suggest either:

* `NavigationServiceDataSource`: The data source pattern is familiar to Cocoa developers.
* `RouteCalculating`: Naming protocols as participles seems to be in vogue these days among Swift developers.

The ideal name would actually be “Routing”, since protocol names shouldn’t be nouns if we can help it, but that’s still too close to Router for comfort.

I’m assuming there’s a gnarly reason we aren’t just folding these methods into the existing NavigationServiceDelegate protocol, but if not, then that would be a lightweight option too.

Reply:
Yeah, naming is hard :)
My current idea is that such objects "provide routing results" so I renamed the protocol. And for our "suggested" implementation I decided to do the same as we did with [Mapbox]NavigationService...